File: Seapal_Gruppe_09/de.htwg.seapal-master/de.htwg.seapal-master/Seapal_php/js/app/ajax/boatinfo.js
$(function() {
	
	function loadEntry(boatnr) { 
			        	
	    jQuery.get("app_boatinfo_load.php", {'bnr': boatnr}, function(data) {

	        $('#bootname').val(data['bootname']);
	        $('#typ').val(data['typ']);
	        $('#baujahr').val(data['baujahr']);
	        $('#registernummer').val(data['registernummer']);
	        $('#konstrukteur').val(data['konstrukteur']);
	        $('#motor').val(data['motor']);
	        $('#segelzeichen').val(data['segelzeichen']);
	        $('#laenge').val(data['laenge']);
	        $('#tankgroesse').val(data['tankgroesse']);
	        $('#heimathafen').val(data['heimathafen']);
	        $('#breite').val(data['breite']);
	        $('#wassertankgroesse').val(data['wassertankgroesse']);
	        $('#yachtclub').val(data['yachtclub']);
	        $('#tiefgang').val(data['tiefgang']);
	        $('#abwassertankgroesse').val(data['abwassertankgroesse']);
	        $('#eigner').val(data['eigner']);
	        $('#masthoehe').val(data['masthoehe']);
	        $('#grosssegelgroesse').val(data['grosssegelgroesse']);
	        $('#versicherung').val(data['versicherung']);
	        $('#verdraengung').val(data['verdraengung']);
	        $('#genuagroesse').val(data['genuagroesse']);
	        $('#rufzeichen').val(data['rufzeichen']);
	        $('#rigart').val(data['rigart']);
	        $('#spigroesse').val(data['spigroesse']);
	
	    } , "json");
	}
	
	function addEntry(bnr, json) {
		
		var entry = ""; 
			
		entry += "<tr class='selectable' id='" + bnr + "'>";
		entry += "<td>" + json.bootname + "</td>";
		entry += "<td>" + json.typ + "</td>";
        entry += "<td>" + json.konstrukteur + "</td>";
        entry += "<td>" + json.baujahr + "</td>";
        entry += "<td>" + json.heimathafen + "</td>";
        entry += "<td>" + json.laenge + "</td>";
        entry += "<td>" + json.breite + "</td>";
        entry += "<td>" + json.tiefgang + "</td>";
        entry += "<td>" + json.eigner + "</td>";
        entry += "<td style='width:30px; text-align:left;'><div class='btn-group'>";
        entry += "<a class='btn btn-small view boat' id='" + bnr + "'><span><i class='icon-eye-open'></i></span></a>";
        entry += "<a class='btn btn-small remove boat' id='" + bnr + "'><span><i class='icon-remove'></i></span></a>";
        entry += "</div></td>";
        entry += "</tr>";

		$('#entries_boat').append(entry);
	}

	$('a.view.boat').live("click", function(event) {
		loadEntry($(this).attr('id'));
	});

	$('a.remove.boat').live("click", function(event) {
		var buttonID = this;
	 	var boatnr = $(this).attr('id');
		jQuery.post("app_boatinfo_delete.php", { "bnr": boatnr }, function(data) {
		 
		 	if (data['bnr'].match(/Error/)) {
		    	
		    	$('#dialogTitle').text('Error');
		    	$('#dialogMessage').text(data['bnr'].replace(/Error: /, ""));
		    	
	    	} else {
		    	
		    	$(buttonID).parents('tr').remove();  
	    
		    	$('#dialogTitle').text('Succes');
		    	$('#dialogMessage').text("Eintrag wurde erfolgreich gelöscht.");
	    	}
			
			$('#messageBox').modal('show');
		}, "json");
	});

	$('#save').click(function(event) {
	
		event.preventDefault();
		
		var json = {
	          "bootname": $('#bootname').val(),
	          "typ": $('#typ').val(),
	          "baujahr": $('#baujahr').val(),
		        "registernummer": $('#registernummer').val(),
		        "konstrukteur": $('#konstrukteur').val(),
		        "motor": $('#motor').val(),
		        "segelzeichen": $('#segelzeichen').val(),
		        "laenge": $('#laenge').val(),
		        "tankgroesse": $('#tankgroesse').val(),
		        "heimathafen": $('#heimathafen').val(),
		        "breite": $('#breite').val(),
		        "wassertankgroesse": $('#wassertankgroesse').val(),
		        "yachtclub": $('#yachtclub').val(),
		        "tiefgang": $('#tiefgang').val(),
		        "abwassertankgroesse": $('#abwassertankgroesse').val(),
		        "eigner": $('#eigner').val(),
		        "masthoehe": $('#masthoehe').val(),
		        "grosssegelgroesse": $('#grosssegelgroesse').val(),
		        "versicherung": $('#versicherung').val(),
		        "verdraengung": $('#verdraengung').val(),
		        "genuagroesse": $('#genuagroesse').val(),
		        "rufzeichen": $('#rufzeichen').val(),
		        "rigart": $('#rigart').val(),
		        "spigroesse": $('#spigroesse').val()              
	    };
		
		jQuery.post("app_boatinfo_insert.php", json, function(data) { 
			
			if (data['bnr'].match(/Error/)) {
		    	
		    	$('#dialogTitle').text('Error');
		    	$('#dialogMessage').text(data['bnr'].replace(/Error: /, ""));
		    	
	    	} else {
		    	
		    	addEntry( data['bnr'], json ); 
	    
		    	$('#dialogTitle').text('Success');
		    	$('#dialogMessage').text("Eintrag wurde erfolgreich gespeichert.");
	    	}
			
			$('#messageBox').modal('show');
		
		}, "json");
	
	});

});

File: Seapal_Gruppe_09/de.htwg.seapal.detp-master/Seapal/app/controllers/Boatinfo.java
package controllers;

import play.*;
import play.mvc.*;
import play.db.*;
import java.sql.*;
import javax.sql.*;
import play.libs.Json;
import play.data.DynamicForm;
import org.codehaus.jackson.node.ObjectNode; 
import views.html.*;
import views.html._include.*;

public class Boatinfo extends Controller {
  
  public static Result insert() {
  
    DynamicForm data = form().bindFromRequest();
    Connection conn = DB.getConnection();
		Statement query;            
    ResultSet result;
    ObjectNode respJSON = Json.newObject();
    int nextId = 0;

    try {
	      query = conn.createStatement();

        query.execute("INSERT INTO seapal.bootinfo(bootname, registernummer, segelzeichen, heimathafen, yachtclub, eigner, versicherung,"
                + "rufzeichen, typ, konstrukteur, laenge, breite, tiefgang, masthoehe, verdraengung, rigart,"
                + "baujahr, motor, tankgroesse, wassertankgroesse, abwassertankgroesse, grosssegelgroesse,"
                + "genuagroesse, spigroesse) "
                + "VALUES('" + data.get("bootname") + "',"
                + "'" + data.get("registernummer") + "',"
                + "'" + data.get("segelzeichen") + "',"
                + "'" + data.get("heimathafen") + "',"
                + "'" + data.get("yachtclub") + "',"
                + "'" + data.get("eigner") + "',"
                + "'" + data.get("versicherung") + "',"
                + "'" + data.get("rufzeichen") + "',"
                + "'" + data.get("typ") + "',"
                + "'" + data.get("konstrukteur") + "',"
                + "'" + data.get("laenge") + "',"
                + "'" + data.get("breite") + "',"
                + "'" + data.get("tiefgang") + "',"
                + "'" + data.get("masthoehe") + "',"
                + "'" + data.get("verdraengung") + "',"
                + "'" + data.get("rigart") + "',"
                + "'" + data.get("baujahr") + "',"
                + "'" + data.get("motor") + "',"
                + "'" + data.get("tankgroesse") + "',"
                + "'" + data.get("wassertankgroesse") + "',"
                + "'" + data.get("abwassertankgroesse") + "',"
                + "'" + data.get("grosssegelgroesse") + "',"
                + "'" + data.get("genuagroesse") + "',"
                + "'" + data.get("spigroesse") + "');");

         result = query.executeQuery("SHOW TABLE STATUS FROM seapal LIKE 'bootinfo'");
         if (result.next()) {
             nextId = result.getInt("Auto_increment");
         }
         conn.close();

         respJSON.put("bnr", "" + (nextId - 1));

    } catch (Exception e) {
        respJSON.put("bnr", "Error: " + e);
    }

    return ok(respJSON);
  }
  
  public static Result delete(int bnr) {

    Connection conn = DB.getConnection();
		Statement query;            
    ResultSet result;
    ObjectNode respJSON = Json.newObject();
  
    try {
	      query = conn.createStatement();
        query.execute("DELETE FROM seapal.bootinfo WHERE bnr = " + bnr);

        conn.close();

        respJSON.put("bnr", "ok");

    } catch (Exception e) {
        respJSON.put("bnr", "Error: " + e);
    }
  
    return ok(respJSON);
  }
  
  public static Result load(int bnr) {
  
    Connection conn = DB.getConnection();
		Statement query;
    ResultSet result;
    ObjectNode respJSON = Json.newObject();

		if(conn != null)
		{
        try {
            	
	          query = conn.createStatement();
	 
	          String sql = "SELECT * FROM seapal.bootinfo WHERE bnr = " + bnr;
	        
	          result = query.executeQuery(sql);
            java.sql.ResultSetMetaData rsmd = result.getMetaData();
            int numColumns = rsmd.getColumnCount();

            while (result.next()) {
                for (int i = 1; i < numColumns + 1; i++) {
                    String columnName = rsmd.getColumnName(i);
                    respJSON.put(columnName, result.getString(i));
                }
            }
            conn.close();

        } catch (Exception e) {
	    	   e.printStackTrace();
        }
    }
    return ok(respJSON);
  }

  public static Result index() {
	    
	  String data = loadEntries();
		
	  return ok(boatinfo.render(header.render(), 
    						    navigation.render("app_map"), 
    						    navigation_app.render("app_boatinfo"), 
    						    data));
  }

	public static String loadEntries() {
	
		Connection conn = DB.getConnection();
		
		String data = "";
    
		if(conn != null)
		{
            Statement query;
            ResultSet result;
            
            try {
            	
	            query = conn.createStatement();
	 
	            String sql = "SELECT * " + "FROM bootinfo ";
	        
	            result = query.executeQuery(sql);
	        
	            while (result.next()) {
              
	        		  StringBuilder row = new StringBuilder();

                row.append("<tr class='selectable' id='" + result.getString("bnr") + "'>");
                row.append("<td>" + result.getString("bootname") + "</td>");
                row.append("<td>" + result.getString("typ") + "</td>");
                row.append("<td>" + result.getString("konstrukteur") + "</td>");
                row.append("<td>" + result.getString("baujahr") + "</td>");
                row.append("<td>" + result.getString("heimathafen") + "</td>");
                row.append("<td>" + result.getString("laenge") + "</td>");
                row.append("<td>" + result.getString("breite") + "</td>");
                row.append("<td>" + result.getString("tiefgang") + "</td>");
                row.append("<td>" + result.getString("eigner") + "</td>");
                row.append("<td style='width:30px; text-align:left;'><div class='btn-group'>");
                row.append("<a class='btn btn-small view' id='" + result.getString("bnr")
                  + "'><span><i class='icon-eye-open'></i></span></a>");
                row.append("<a class='btn btn-small remove' id='" + result.getString("bnr")
                  + "'><span><i class='icon-remove'></i></span></a>");
                row.append("</div></td>");

                row.append("</tr>");
            
		            data += row.toString();
			    }
               
	       } catch (Exception e) {
	    	   e.printStackTrace();
	       }
        }
            
        return data;
	}

  
}